node/init.go

```diff
@@ -48,6 +49,10 @@ func InitWith(path string, tp Type, cfg *Config) error {
 		return err
 	}
 
+	if cfg == nil {
+		return errors.New("nil config provided")
```

Comment:
I suppose `nil config provided` is not telling the user enough information on what went wrong during initialisation phase. Let's have something like this
```suggestion
		return errors.New("Configuration is missing for the node's initialisation")
```